src/status.rs

```diff
@@ -0,0 +1,95 @@
+//! A Status Report announces the station's current mission or any other single
+//! line status to everyone. The report starts with the '>' APRS Data Type Identifier.
+//! The report may optionally contain a timestamp.
+//!
+//! Examples:
+//! - ">12.6V 0.2A 22degC"              (report without timestamp)
+//! - ">120503hFatal error"             (report with timestamp in HMS format)
+//! - ">281205zSystem will shutdown"    (report with timestamp in DHM format)
+
+use std::convert::TryFrom;
+use std::io::Write;
+
+use AprsError;
+use EncodeError;
+use Timestamp;
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct AprsStatus {
+    pub timestamp: Option<Timestamp>,
+    pub comment: Vec<u8>,
+}
+
+impl TryFrom<&[u8]> for AprsStatus {
+    type Error = AprsError;
+
+    fn try_from(b: &[u8]) -> Result<Self, Self::Error> {
+        // parse timestamp if available
+        let has_timestamp = b.len() >= 7
+            && (b[6] == b'h' || b[6] == b'z')
+            && b[0..6].iter().all(|c| c.is_ascii_digit());
+        let timestamp = if has_timestamp {
+            Some(Timestamp::try_from(
+                b.get(0..7)
+                    .ok_or_else(|| AprsError::InvalidPosition(b.to_vec()))?,
```

Comment:
Not sure if we want to reuse `AprsError::InvalidPosition` here. Might be better to introduce a new error type

Reply:
Thank you, just fixed it.